### PR TITLE
Adding Material-ui Examples

### PR DESCRIPTION
**Pull request by mihirsoni**
@erikras  

As discussed I have tried to put things in following way.

- Added simple form for material-ui example , I will try to add more.

Things I have observed. 

- With material-ui it is difficult to use as stateless components.
- For for some controls like ``SelectField`` we are treating as custom Component and we are firing onChange manually. Please let me know if there could be better way.

### CODE REVIEW

**Comment by erikras** [comment]
No, no, I want the dir structure to look like this:
```
examples/asyncValidation
examples/immutable
examples/initializeFromState
examples/material-ui       <----- your new example
examples/simple
examples/submitValidation
examples/syncValidation
examples/wizard
```

**Comment by mihirsoni** [comment]
@erikras  done. Reverted the directory changes.

File: examples/material-ui/README.md [hunk 1]
```diff
@@ -0,0 +1,15 @@
+# Examples
```
Comment (erikras) [inline examples/material-ui/README.md]: This `README.md` should be like its siblings. e.g. `examples/simple/README.md`

**Comment by erikras** [comment]
The `simple` dir under `material-ui` is superfluous.
```bash
cd examples/material-ui
mv simple/* .
rmdir simple
```

File: examples/material-ui/README.md (hunk 1 above)
Comment (mihirsoni) [inline examples/material-ui/README.md]: @erikras  okay then we don't need to replicate all the examples in material UI. Have I understood correctly ?

**Comment by mihirsoni** [comment]
@erikras it should be fixed now. Sorry for confusion. I misunderstood.

File: examples/material-ui/index.html [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+  <title>Redux Form</title>
+  <link href="http://redux-form.com/6.0.0-alpha.6/bundle.css"
+    media="screen, projection"
+    rel="stylesheet" type="text/css"/>
+  <link href="//cdnjs.cloudflare.com/ajax/libs/font-awesome/4.3.0/css/font-awesome.min.css"
+    media="screen, projection" rel="stylesheet" type="text/css"/>
+   <style type="text/css">
+   /*This CSS has been tweked from root site 
+      to match with material-ui*/
+    form > div > div > div {
```
Comment (mihirsoni) [inline examples/material-ui/index.html]: @erikras This is the simple override of css in this file. If UI looks good to you we don't need to change anything.

**Comment by erikras** [comment]
`SimpleForm.js` --> `MaterialUiForm`
`Simple.md` --> `MaterialUi.md`

**Comment by mihirsoni** [comment]
@erikras It is done now. Thank you for the review.

**Comment by erikras** [comment]
I saw. 👍 I'll test it later today.

**Comment by erikras** [comment]
Looks good! Two things, for another PR:

1. Please add validation errors (I wrote the validation function already) to display however Material UI does it
2. `npm run build` never completes for me. Webpack is unhappy. See if you can figure that out.

**Comment by mihirsoni** [comment]
@erikras  Sure , I will do the validation.

`npm  run build` works fine at my place

```sh
$ npm run build

> redux-form-simple-material-ui-example@6.0.0-alpha.6 build /Users/mihir/Documents/redux-form-forked/redux-form/examples/material-ui
> npm run clean && npm run build:webpack


> redux-form-simple-material-ui-example@6.0.0-alpha.6 clean /Users/mihir/Documents/redux-form-forked/redux-form/examples/material-ui
> rimraf dist


> redux-form-simple-material-ui-example@6.0.0-alpha.6 build:webpack /Users/mihir/Documents/redux-form-forked/redux-form/examples/material-ui
> cross-env NODE_ENV=production webpack --config webpack.config.prod.js

Hash: c578edfe4bd84fcf8626
Version: webpack 1.13.0
Time: 99594ms
        Asset    Size  Chunks             Chunk Names
    bundle.js  760 kB       0  [emitted]  main
bundle.js.map  5.1 MB       0  [emitted]  main
   [0] multi main 40 bytes {0} [built]
    + 658 hidden modules
```

**Comment by erikras** [comment]
What version of `npm` are you using?

**Comment by erikras** [comment]
Apparently I just wasn't patient enough. Sheesh!

material-ui -fish fish 158x32 5 2016-04-29 18-27-37

**Comment by mihirsoni** [comment]
@erikras  Yes even I noticed this issue , not able to figure out why the hell it takes this much time 😢 I will try to figure this out.

**Pull request merged**